tf2onnx/optimizer/transpose_optimizer.py

```diff
@@ -239,10 +242,14 @@ def _switch_transpose_and_node(self, node, trans):
         # need to transpose node shape in backward direction as well after switch
         # otherwise, reshape added in post_optimize_action may not work correctly
         shape = self._g.get_shape(node.output[0])
-        if shape:
+        self._g.set_shape(trans.output[0], shape)
+        if shape and len(shape) == 4:

```

Comment:
when will the rank !=4?
could you add more information about the bug to the PR description?

Reply:
done.
for example: transpose following with a squeeze.